### PR TITLE
analytics/withdraw v1

### PR DESCRIPTION
**Pull request by pedroapfilho**
- feat(analytics): adding types for withdrawal_v1
- feat(analytics): adding remaining withdraw events
- feat(analytics): adding link bank clicked event

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts [hunk 1]
```diff
@@ -263,6 +263,24 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
...
+
+    const bankTransferAccountsR = selectors.components.brokerage.getBankTransferAccounts(
+      yield select()
+    )
+    const bankTransferAccounts = bankTransferAccountsR.getOrElse([])
+    if (bankTransferAccounts.length) {
+      yield put(
+        actions.components.brokerage.setBankDetails({
+          account: bankTransferAccounts.filter(
+            (a) => a.currency === payload.fiatCurrency && a.state === 'ACTIVE'
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:277]: maybe move 'ACTIVE' to be enum

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts [hunk 2]
```diff
@@ -99,6 +112,12 @@ type PageViewPayload = {
   url: string
 }
 
+type PageNamesType = '/home'
+// | '/interest'
+// | '/settings/general'
+// | '/settings/preferences'
+// | '/settings/addresses'
+
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts:120]: should we delete this completely?

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts [hunk 3]
```diff
@@ -163,6 +182,10 @@ type EmailVerificationClickedPayload = BasePayload & {
   // origin: 'SIGN_UP' | 'VERIFICATION'
 }
 
+type LinkBankClickedPayload = BasePayload & {
+  // origin: 'BUY' |'DEPOSIT' | 'SETTINGS' | 'WITHDRAW'
+}
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts:187]: you can remove this part :)

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts [hunk 4]
```diff
@@ -308,6 +331,33 @@ type UpgradeVerificationClickedPayload = BasePayload & {
...
+  withdrawal_method: WithdrawalMethodType
+}
+type WithdrawalAmountMinClickedPayload = BasePayload & {
+  currency: string
+  withdrawal_method: WithdrawalMethodType
+}
+
+type WithdrawalClickedPayload = BasePayload & {
+  // origin: 'CURRENCY_PAGE' | 'PORTFOLIO'
+}
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts:352]: clean up 👍

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts (hunk 2 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts:120]: I'm going to use it on the next PRs

File: packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Withdraw/EnterAmount/index.tsx [hunk 5]
```diff
@@ -20,96 +20,89 @@ import Failure from './template.failure'
...
-      )
-    })
-  }
+  return props.data.cata({
+    Failure: () => <Failure {...props} />,
+    Loading: () => <Loading />,
+    NotAsked: () => <Loading />,
+    Success: (val) => (
+      <Success
+        {...props}
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Withdraw/EnterAmount/index.tsx:99]: We do this all around app, but would be good practice to stop it and prevent spreading all props from top down, better to pass only necessary props.

**Review by milan-bc — APPROVED** [review]
Great work so far @pedroapfilho I left few minor comments, nothing blocker to be merged.

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts (hunk 1 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:277]: I'm not exactly sure how it works. I've just changed it for the linter.

File: packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Withdraw/EnterAmount/index.tsx (hunk 5 above)
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Withdraw/EnterAmount/index.tsx:99]: Why?

**Pull request merged**